### PR TITLE
(WIP, Do not Merge) : Make /serial suffix applicable for neuron and hdf5

### PR DESCRIPTION
**Pull request by pramodk**
- #371 added /serial suffix for ~mpi variant which
    is not strict
  - we want to add serial suffix for neuron~mpi only

Didn't realise `/serial` is appended everywhere:

image

Note : we should this PR after carefully checking module names generated in PR. I wonder if will end-up **with and without /serial** (i.e. duplicates) due to rsync.

### CODE REVIEW

**Review by matz-e — APPROVED** [review]

**Comment by pramodk** [comment]
NOTE: We shouldn't merge this until generated modules are reviewed !

**Comment by matz-e** [comment]
> NOTE: We shouldn't merge this until generated modules are reviewed !

That's what *draft* pull requests are for ;)

File: deploy/configs/parallel-libraries/modules.yaml
Comment (matz-e) [inline deploy/configs/parallel-libraries/modules.yaml:48]: I think this could stay.

**Comment by pramodk** [comment]
Please retest

**Comment by pramodk** [comment]
This gives:

```
unset MODULEPATH
PULL_ID=497
for tag in "compilers" "tools" "external-libraries" "serial-libraries"  "parallel-libraries" "applications";
do
    export MODULEPATH=/gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/497/modules/$tag/2019-09/tcl/linux-rhel7-x86_64:$MODULEPATH;
done

kumbhar@bbpv1:~$ module av

--------------------------------------------- /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/497/modules/applications/2019-09/tcl/linux-rhel7-x86_64 ----------------------------------------------
archngv/0.0.0/python3                    neurodamus-hippocampus/0.4/python3       py-bluepy/0.12.7/python3                 py-entity-management/1.0.9/python3
asciitoh5/1.0/python3/serial             neurodamus-mousify/0.3/python3           py-bluepy/0.13.0/python3                 py-jupyter-notebook/4.2.3/python3
brainbuilder/0.11.3/python3              neurodamus-neocortex/0.3/knl/python3     py-bluepy/0.13.3/python2                 py-pytouchreader/1.4.3/python3
brainbuilder/0.11.4/python3              neurodamus-neocortex/0.3/python3         py-bluepy/0.13.3/python3                 py-simwriter/0.4.3/python2
brayns/0.8.0                             neurodamus-thalamus/0.3/knl/python3      py-bluepy/0.13.5/python2                 reportinglib/2.5.1
brayns/immersive                         neurodamus-thalamus/0.3/python3          py-bluepy/0.13.5/python3                 reportinglib/develop
brayns-brainatlas/0.1.0                  parquet-converters/0.4.1                 py-bluepymm/0.6.38/python2/serial        spatial-index/0.1.0/python3
brayns-topologyviewer/0.1.0              placement-algorithm/2.0.8/python3        py-bluepymm/0.6.38/python3/serial        spykfunc/0.14.3/python3
brion/3.0.0/python3                      psp-validation/0.1.12/python3/serial     py-bluepyopt/1.6.56/python2/serial       steps/3.4.1/python3
connectome-tools/0.3.2/python3           py-basalt/0.2.4/python3                  py-bluepyopt/1.6.56/python3/serial       synapsetool/0.5.2
functionalizer/3.12.2                    py-bbp-analysis-framework/0.6.39/python2 py-efel/3.0.22/python2                   touchdetector/4.4.2
neurodamus-core/2.7.0/python3            py-bglibpy/4.1.4/python2/serial          py-efel/3.0.22/python3                   touchdetector/5.1.0
neurodamus-hippocampus/0.4/knl/python3   py-bglibpy/4.1.4/python3/serial          py-entity-management/1.0.9/python2       touchdetector/5.2.0

------------------------------------------ /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/497/modules/parallel-libraries/2019-09/tcl/linux-rhel7-x86_64 -------------------------------------------
hdf5/1.10.4             highfive/2.0            neuron/7.6.8/python3    omega-h/9.29.0          petsc/3.10.2            py-mpi4py/3.0.0/python3 trilinos/xsdk-0.4.0-rc1

------------------------------------------- /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/497/modules/serial-libraries/2019-09/tcl/linux-rhel7-x86_64 --------------------------------------------
boost/1.65.0/python2             gmsh/3.0.6                       highfive/2.0                     neuron/7.6.8/python3/serial      python-dev/0.2/python2/serial
boost/1.65.0/python3             hdf5/1.10.4/serial               highfive/develop/serial          python-dev/0.1/python2/serial    python-dev/0.2/python3/serial
boost/1.68.0                     highfive/1.6                     neuron/7.6.8/python2/serial      python-dev/0.1/python3/serial    tensorflow/1.12.0/python3/serial

------------------------------------------ /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/497/modules/external-libraries/2019-09/tcl/linux-rhel7-x86_64 -------------------------------------------
cuda/9.0.176            cudnn/7.3               python/2.7.15           snakemake/5.6.0/python3
cuda/9.2.88             julia/1.2.0-rc1/python3 python/3.6.5            spark/2.4.0

------------------------------------------------- /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/497/modules/tools/2019-09/tcl/linux-rhel7-x86_64 -------------------------------------------------
allinea-forge/19.0.5 doxygen/1.8.15       ffmpeg/3.2.4         hpe-mpi/2.16         ispc/1.10.0          valgrind/3.13.0
cmake/3.13.0         emacs/26.1           ffmpeg/4.2           intel-mpi/2018.1.163 mvapich2/2.3

----------------------------------------------- /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/497/modules/compilers/2019-09/tcl/linux-rhel7-x86_64 -----------------------------------------------
gcc/6.4.0                            gcc/8.2.0                            intel-parallel-studio/cluster.2018.3 pgi/19.4
gcc/7.3.0                            intel/18.0.3                         llvm/7.0.0
```

We are wondering if 
```
    suffixes:
        'hdf5~mpi': '/serial'
```
is being applied to python-dev as well (like `^hdf5~mpi` which is weird).

**Review by ferdonline — APPROVED** [review]

**Review by matz-e — CHANGES_REQUESTED** [review]
@pramodk before anyone else merges this, can this just be closed now?

**Comment by pramodk** [comment]
Cleanup!

**Pull request closed without merging**
